docs/source/notes/autograd.rst

```diff
@@ -210,3 +210,79 @@ No thread safety on C++ hooks
 Autograd relies on the user to write thread safe C++ hooks. If you want the hook
 to be correctly applied in multithreading environment, you will need to write
 proper thread locking code to ensure the hooks are thread safe.
+
+Autograd for Complex Numbers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+What notion of complex derivative does PyTorch use?
+***************************************************
+
+   PyTorch follows `_JAX's <https://jax.readthedocs.io/en/latest/notebooks/autodiff_cookbook.html#Complex-numbers-and-differentiation>`
+   convention for autograd for Complex Numbers.
+
+    For a function :math:`F: C → C`
+
+    .. math::
+        def F(z):
+            x, y = real(z), imag(z)
+            return u(x, y) + v(x, y) * 1j
+
+    The JVP and VJP for function :math:`F` are defined as:
+
+    .. math::
+
+        J = \begin{bmatrix}
+            \partial_0u(x, y) & \partial_1u(x, y)\\
+            \partial_0v(x, y) & \partial_1v(x, y) \end{bmatrix}
+
+        def JVP(tangent):
+            c, d = real(tangent), imag(tangent)
+            return \begin{bmatrix} 1 & i \end{bmatrix} * J * \begin{bmatrix} c \\ d \end{bmatrix}
+
+        def VJP(cotangent):
+            c, d = real(cotangent), imag(cotangent)
+            return \begin{bmatrix} c & -d \end{bmatrix} * J * \begin{bmatrix} 1 \\ -i \end{bmatrix}
```

Comment:
In PyTorch, the VJP is mostly what we care about, as it is the computation performed when we do backwards mode automatic differentiation. Notice that d and i are negated in the formula above.

Reply:
I might suggest absorbing "How are the JVP and VJP defined for :math:`R^2 -> C` and :math:`C -> R^2` functions?" section into this one. The structure then is "Here is the general definition", and then "Here is a particular example"

Reply:
I just felt it would be cleaner to have them in separate sections primarily because not everyone looking to get information about autograd for complex functions would be interested in cross domain functions